File: tests/true_e2e/fixtures/broken_c/broken.c
/**
 * RCH E2E Test Fixture - Broken C Code
 *
 * This file contains intentional compilation errors for testing
 * exit code 1 propagation from gcc/clang.
 */

#include <stdio.h>

int main(void) {
    // Intentional error: undeclared variable
    int result = undefined_function(42);

    // Intentional error: type mismatch
    char* str = 42;  // warning that becomes error with -Werror

    // Intentional error: missing semicolon
    printf("This will not compile")

    return result;
}

File: tests/true_e2e/fixtures/broken_rustc/broken.rs
// RCH E2E Test Fixture - Broken Rust File
//
// This file contains intentional compilation errors for testing
// exit code 1 propagation from direct rustc invocation.

fn main() {
    // Intentional error: type mismatch
    let x: i32 = "not an integer";

    // Intentional error: undefined function
    let y = undefined_function();

    // Intentional error: missing semicolon
    println!("This will not compile")
    let z = 42;

    println!("{} {} {}", x, y, z);
}
